### PR TITLE
update build notification email docs

### PR DESCRIPTION
**Pull request by angusfretwell**
Back in the olden days of 2022, we [extended email notifications](https://github.com/buildkite/buildkite/pull/8696) to send on `build_failing` as well as `build_finished`. 

@sj26 [predicted](https://github.com/buildkite/buildkite/pull/8696#issuecomment-1123152714) that customers would ask:
>how do I opt out of failing notifications, if I only want finished? 

That prediction has been proved correct by [this incident](https://coda.io/d/Escalations-Feedback_dHnUHNps1YO/Escalations_sudOj#escalations-board_tu9Bj/r213&view=modal). The customer was rightly confused as the docs are currently misleading.

This PR updates the documentation to explain that emails send on `build_failing` as well as `build_finished`, and demonstrates how to filter the notification if you only want to receive `build_finished` emails.

### CODE REVIEW

File: pages/pipelines/notifications.md [hunk 1]
```diff
@@ -60,7 +60,17 @@ notify:
 ```
 {: codeblock-file="pipeline.yml"}
 
-You can only send email notifications on entire pipeline events, specifically upon [`build finished`](/docs/apis/webhooks#events).
+You can only send email notifications on entire pipeline events, specifically upon [`build_failing`](/docs/apis/webhooks#events) and [`build finished`](/docs/apis/webhooks#events).
```
Comment (angusfretwell) [inline pages/pipelines/notifications.md:63]:
```suggestion
You can only send email notifications on entire pipeline events, specifically upon [`build_failing`](/docs/apis/webhooks#events) and [`build_finished`](/docs/apis/webhooks#events).
```

**Review by jordandcarter — APPROVED** [review]
LGTM! 🚀

**Review by gilesgas — APPROVED** [review]

File: pages/pipelines/notifications.md [hunk 2]
```diff
@@ -58,7 +58,17 @@ notify:
...
 
-You can only send email notifications on entire pipeline events, specifically upon [`build finished`](/docs/apis/webhooks#events).
+You can only send email notifications on entire pipeline events, specifically upon [`build_failing`](/docs/apis/webhooks#events) and [`build_finished`](/docs/apis/webhooks#events).
+
+Restrict notifications to finished builds by adding a [conditional](#conditional-notifications):
+
+```yaml
+notify:
+  - email: "dev@acmeinc.com"
+    if: build.state != "failing"
```
Comment (gilesgas) [inline pages/pipelines/notifications.md:68]: In this example, I'm assuming this value doesn't need to be `build.failing`?

File: pages/pipelines/notifications.md [hunk 3]
```diff
@@ -60,7 +60,17 @@ notify:
 ```
 {: codeblock-file="pipeline.yml"}
 
-You can only send email notifications on entire pipeline events, specifically upon [`build finished`](/docs/apis/webhooks#events).
+You can only send email notifications on entire pipeline events, specifically upon [`build_failing`](/docs/apis/webhooks#events) and [`build_finished`](/docs/apis/webhooks#events).
```
[the remark this answers is not in the document]
Comment (gilesgas) [reply pages/pipelines/notifications.md]: Thanks for confirming Angus :-)

**Pull request merged**